### ISSUE TITLE
Proxy Logic is a little off

### ISSUE THREAD

**Issue by kellya** [PokemonGoMap/PokemonGo-Map#1632]
##Summary

When using proxies, it is requring the number of proxies to match the number of workers..

## Expected Behavior
If only one proxy is specified, it should use the same proxy for all workers

## Current Behavior
If you specify fewer proxies than you have workders, you will get an index out of range error.  This seems to be because the logic in pogom/proxy.py keeps incrementing last_proxy past len(args.proxy).

## Possible Solution
To me, the logic looks correct, and I wasn't able to write anything that worked, So I ended up changing the return line to this:
```
if lp < len(args.proxy):
        return lp, args.proxy[lp]
    else:
        return lp, args.proxy[0]
```
Which works, but is extra logic that shouldn't be there.
## Steps to Reproduce (for bugs)
Using a config file:
1. use workers : 3
2. use a single proxy with proxy: socks5://127.0.0.1:9050
3.  Once you run runserver.py, you will get index errors

## Context
This happens once it starts to create the search workers. It will attempt to get next proxy, will increment last_proxy past the index range and throw the error.

## Your Environment
* Version used:git 516c2160d7f
* Python 2.7.12 (under virtualenv)
* Fedora 25

**Comment by kellya** [comment]
I messed with greater than, less than, reversing the logic and ultimately my work-around I posted here.  Subtracting 1 from the length eluded me :)

**Comment by birdstream** [comment]
I was scratching my head for a bit, too.. I'm not a programmer. Haven't really been programming since i was a kid messing around in BASIC on my C64 (oh the times!).
Anyway, I just added a line to print all relevant variables to logging, and I immediately spotted the problem :)

**Comment by sebastienvercammen** [comment]
Merged #1619
